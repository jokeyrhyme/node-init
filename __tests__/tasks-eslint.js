/* @flow */
'use strict';

jest.mock('fs');

jest.mock('execa');
jest.mock('update-json-file');

test('fn()', async () => {
  const { fn } = require('../lib/tasks/eslint.js');
  await fn({ cwd: process.cwd(), scope: '' });

  expect(require('execa').mock.calls).toMatchSnapshot();

  // $FlowFixMe: ignore .mock's missing type
  expect(require('update-json-file').mock.calls.length).toBe(2);

  // $FlowFixMe: ignore .mock's missing type
  for (const [filePath, editor] of require('update-json-file').mock.calls) {
    expect(filePath).toMatchSnapshot();
    expect(editor({})).toMatchSnapshot();
  }
});
