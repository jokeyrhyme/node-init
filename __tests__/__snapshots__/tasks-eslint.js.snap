// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn() 1`] = `
Array [
  Array [
    "npm",
    Array [
      "uninstall",
      "--save-dev",
      "eslint-config-semistandard",
      "eslint-config-standard",
      "eslint-config-standard-jsx",
      "eslint-config-standard-react",
      "eslint-plugin-standard",
    ],
    Object {
      "cwd": "/home/ron/GitHub/jokeyrhyme/node-init.js",
    },
  ],
  Array [
    "npm",
    Array [
      "install",
      "--save-dev",
      "eslint",
      "eslint-plugin-import",
      "eslint-plugin-node",
      "eslint-plugin-promise",
    ],
    Object {
      "cwd": "/home/ron/GitHub/jokeyrhyme/node-init.js",
    },
  ],
]
`;

exports[`fn() 2`] = `"/home/ron/GitHub/jokeyrhyme/node-init.js/package.json"`;

exports[`fn() 3`] = `
Object {
  "scripts": Object {
    "eslint": "eslint --fix --cache .",
    "pretest": "npm run eslint",
  },
}
`;

exports[`fn() 4`] = `"/home/ron/GitHub/jokeyrhyme/node-init.js/package.json"`;

exports[`fn() 5`] = `
Object {
  "eslintIgnore": Array [
    "build",
    "coverage",
    "dist",
  ],
}
`;
